src/patternfly/components/DataList/data-list.scss

```diff
@@ -123,48 +130,49 @@
 
   // Compact
   --pf-c-data-list--m-compact--FontSize: var(--pf-global--FontSize--sm);
+  --pf-c-data-list--m-compact--LineHeight: var(--pf-global--LineHeight--sm);
   --pf-c-data-list--m-compact__check--FontSize: var(--pf-global--FontSize--md);
-  --pf-c-data-list--m-compact__cell--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-data-list--m-compact__cell--PaddingTop: calc(var(--pf-global--spacer--sm) + (var(--pf-c-data-list--LineHeight) * 1px));
   --pf-c-data-list--m-compact__cell--PaddingBottom: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__cell--md--PaddingBottom: 0;
   --pf-c-data-list--m-compact__cell-cell--PaddingTop: 0;
   --pf-c-data-list--m-compact__cell-cell--md--PaddingTop: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__cell--cell--MarginRight: var(--pf-global--spacer--md);
-  --pf-c-data-list--m-compact__item-control--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-data-list--m-compact__item-control--PaddingTop: calc(var(--pf-global--spacer--sm) + (var(--pf-c-data-list--LineHeight) * 1px));
   --pf-c-data-list--m-compact__item-control--PaddingBottom: 0;
   --pf-c-data-list--m-compact__item-control--MarginRight: var(--pf-global--spacer--md);
-  --pf-c-data-list--m-compact__item-action--PaddingTop: var(--pf-global--spacer--sm);
-  --pf-c-data-list--m-compact__item-action--PaddingBottom: var(--pf-global--spacer--sm);
+  --pf-c-data-list--m-compact__item-action--PaddingTop: 0;
+  --pf-c-data-list--m-compact__item-action--PaddingBottom: 0;
+  --pf-c-data-list--m-compact__action--MarginTop: 0;
   --pf-c-data-list--m-compact__item-action--MarginLeft: var(--pf-global--spacer--md);
   --pf-c-data-list--m-compact__item-content--PaddingBottom: var(--pf-global--spacer--sm);
 
   @include pf-t-light;
 
+  font-size: var(--pf-c-data-list--FontSize);
+  line-height: var(--pf-c-data-list--LineHeight);
   overflow-wrap: break-word;
   list-style-type: disc;
   border-top: var(--pf-c-data-list--BorderTopWidth) solid var(--pf-c-data-list--BorderTopColor);
 
   &.pf-m-compact {
-    font-size: var(--pf-c-data-list--m-compact--FontSize);
-
     .pf-c-data-list__cell {
       --pf-c-data-list__cell--PaddingTop: var(--pf-c-data-list--m-compact__cell--PaddingTop);
       --pf-c-data-list__cell--PaddingBottom: var(--pf-c-data-list--m-compact__cell--PaddingBottom);
       --pf-c-data-list__cell--MarginRight: var(--pf-c-data-list--m-compact__cell--cell--MarginRight);
       --pf-c-data-list__cell--cell--PaddingTop: var(--pf-c-data-list--m-compact__cell-cell--PaddingTop);
     }
 
+    --pf-c-data-list--FontSize: var(--pf-c-data-list--m-compact--FontSize);
```

Comment:
nit - can you move these vars before the `.pf-c-data-list__cell` block so the vars come first, then styles?